### PR TITLE
Spinner: Fix so that the `size` property is correctly applied

### PR DESCRIPTION
**Pull request by ashharrison90**
**What is this feature?**

made a slight mistake in https://github.com/grafana/grafana/pull/76819 and broke how sizes were being applied to `<Spinner />`. before, `Spinner` allowed any number (e.g. `16`) or string (e.g. `"16px"`) to be passed as the size, and this was applied as a font size. now that the `Spinner` is a proper svg and not using the font-awesome font, this font size doesn't do anything. 

- deprecates the old sizes (e.g. `16` or `"16px"`) in favour of having the same predefined sizes as `Icon` (e.g. `sm`, `md`, `lg`, etc.)
- adds a new `isIconSize` type guard
  - we can remove this if we want once the old interface is gone
- changes `Spinner` to support both the old syntax and new syntax behind this `isIconSize` type guard
- refactors everywhere in core grafana to use these new standard spinner sizes
- removes applying the `fa-spin` class from the `spinner` icon. instead this class is only applied by the `Spinner` component

**Why do we need this feature?**

- fixes sizes being applied to `Spinner` correctly
- cleaner interface (in the end) for `Icon` and `Spinner`

**Who is this feature for?**

- anyone using `Spinner`

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/features/plugins/sql/components/query-editor-raw/QueryValidator.tsx [hunk 1]
```diff
@@ -79,7 +79,7 @@ export function QueryValidator({ db, query, onValidate, range }: QueryValidatorP
     <>
       {state.loading && (
         <div className={styles.info}>
-          <Spinner inline={true} size={12} /> Validating query...
+          <Spinner inline={true} size="sm" /> Validating query...
```
Comment (L-M-K-B) [inline public/app/features/plugins/sql/components/query-editor-raw/QueryValidator.tsx:82]: 12px would be size xs. Did you change it to sm (14px) on purpose?

**Review by L-M-K-B — COMMENTED** [review]
Looks good so far. I only have one question.

File: public/app/features/plugins/sql/components/query-editor-raw/QueryValidator.tsx (hunk 1 above)
Comment (ashharrison90) [reply public/app/features/plugins/sql/components/query-editor-raw/QueryValidator.tsx]: sure didn't, just a mistake 😄 

great spot! 🚀

**Review by Clarity-89 — APPROVED** [review]
Looks good!

**Review by L-M-K-B — APPROVED** [review]
Alright, it's time to merge then! 🥳

**Pull request merged**